src/tools/connection_patcher/Program.cpp

```diff
@@ -32,6 +32,9 @@
 
 #if PLATFORM == PLATFORM_WINDOWS
 #include <Shlobj.h>
+#elif PLATFORM == PLATFORM_UNIX
```

Comment:
Nope: https://github.com/TrinityCore/TrinityCore/blob/6.x/src/tools/connection_patcher/Program.cpp#L192-L194

commonData variable (which is used as root path for adding bnet path to it) was only being initializated in Windows, so for linux and mac, a exception was being thrown due to it